topcoffea/modules/YieldTools.py

```diff
@@ -0,0 +1,436 @@
+import gzip
+import json
+import pickle
+import numpy as np
+from coffea import hist
+#from topcoffea.modules.HistEFT import HistEFT
+from topcoffea.modules.paths import topcoffea_path
+
+class YieldTools():
+
+    def __init__(self):
+
+        # The path to the file listing the luminosity for each year
+        self.LUMI_FILE = topcoffea_path("json/lumi.json")
+
+        # The order of the categories in the TOP-19-001 AN yield tables
+        self.CAT_LST = ["cat_2lss_p","cat_2lss_m","cat_3l_1b_offZ_p","cat_3l_1b_offZ_m","cat_3l_2b_offZ_p","cat_3l_2b_offZ_m","cat_3l_1b_onZ","cat_3l_2b_onZ","cat_4l"]
+
+        # A dictionary mapping names of samples in the samples axis to a short version of the name
+        self.PROC_MAP = {
+            "ttlnu" : ["ttW_centralUL17","ttlnu_private2017","ttlnuJet_privateUL17","ttlnuJet_privateUL18"],
+            "ttll"  : ["ttZ_centralUL17","ttll_TOP-19-001","ttllJet_privateUL17","ttllJet_privateUL18"],
+            "ttH"   : ["ttH_centralUL17","ttH_private2017","ttHJet_privateUL17","ttHJet_privateUL18"],
+            "tllq"  : ["tZq_centralUL17","tllq_private2017","tllq_privateUL17","tllq_privateUL18"],
+            "tHq"   : ["tHq_central2017","tHq_privateUL17"],
+            "tttt"  : ["tttt_central2017","tttt_privateUL17"],
+        }
```

Comment:
I've been working on a way to dynamically create these. In the datacard maker I have a dictionary called `rename` (though you don't have to call yours this), which is initially defined for things like `ttZ` to `ttll`
https://github.com/TopEFT/topcoffea/blob/ht_cuts/analysis/topEFT/datacard_maker.py#L17
then I dynamically grab the process names from a histogram, and use a regex to extract the important parts, and append this to `rename`
https://github.com/TopEFT/topcoffea/blob/ht_cuts/analysis/topEFT/datacard_maker.py#L41-L42
which produces:
`{'tZq': 'tllq', 'tllq_privateUL17': 'tllq', 'ttZ': 'ttll', 'ttlnu': 'ttlnuJet_privateUL17', 'tttt': 'tttt_privateUL17', 'ttll': 'ttllJet_privateUL17', 'tllq': 'tllq_privateUL17', 'ttH': 'ttHJet_privateUL17', 'tHq': 'tHq_privateUL17'}`
If you're interested, I can help you implement something similar here, so we don't have to keep track of all the different names floating around. The only issues I foresee with this is we must conform to the regex style of `process_label`, or rewrite the regex.

Reply:
Thanks @bryates, this looks interesting. I agree it'd be good to add something like that at some point, but as this PR is really just a preliminary implementation aimed at getting some baseline tools in place, I think I'm fine with leaving the hardcoded dictionary for now. So if it's ok with you, I think I'd probably prefer to just merge what we have at this point, and then we can improve on it in future branches.

Reply:
Sounds good to me! I have no objections if you want to hit merge now.